sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,647 @@
+
+Therefore, this extension currently only supports class 1 of operations because the mapping between `get_wi_data` and `joint_matrix` elements is not required to be known for these operations. However, general piece-wise operations will be supported in the future as a new API will be provided to convey the mapping from `joint_matrix` domain to WI Domain (See Section "WI data to joint matrix mapping coordinates information for piece-wise operations for more information").
+
+Also, note that `get_wi_data` cannot return a fixed size array length because the length of the WI portion is a runtime variable for the following reasons:
+
+1- The main compilation mode of SYCL is JIT compilation and partitioning among WIs is implementation defined.
+
+2- SG size is not generally fixed.
+
+3- AMX has the flexibility of allowing variable sizes on the matrix (`dynamic_extent`).
+
+In the case of CUDA backend which is SYCL AOT compiled and SG size = 32 known and fixed, the additional marray capability will be provided.
+
+The code listing below shows a synopsis of these new APIs.
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+template <typename T, size_t NumRows, size_t NumCols,
+          use U, layout L = layout::unused,
+          typename Group = sycl::sub_group>
+struct joint_matrix {
+   wi_data<T, U, NumRows, NumCols, L, Group> get_wi_data();
+};
+template <typename T, size_t NumRows, size_t NumCols, use U, layout L, typename Group>
+class wi_data {
+  size_t length();
+  wi_element<T, NumRows, NumCols, U, L, Group> operator[](size_t i);
+};
+template <typename T, size_t NumRows, size_t NumCols,
+          use U, layout L = layout::unused,
+          typename Group = sycl::sub_group>
+class wi_element {
+  operator T();
+  wi_element &operator=(const T &rhs);
+…
+};
+}
+```
+
+In the following example `wi_data_c` is a reference to the WI owned portion of the joint matrix `matC`. As such `wi_data_c[i] OP rhs` updates the corresponding matrix element in the joint_matrix `matC`.
+Vectorization along the subgroup dimension will get enabled automatically to vectorize the contiguous portion of the matrix. 
+
+
+```c++
+auto wi_data_c = matC.get_wi_data();
+for (int i = 0; i < wi_data_c.length(); i++)
+        wi_data_c[i] *= alpha;    // Note that the indexing here "i" is in the vector owned by a WI, not in the matrix C        
+```
+
+IMPORTANT: In the current implementation, only the subgroup scope is supported.
+
+IMPORTANT: The WI data to joint matrix mapping coordinates information is not implemented yet.
+
+IMPORTANT: Since the current tensorcores implementation is AOT, it is possible to know how many elements are owned by each WI at compile time. In this case, `wi_data` can be of type `marray`. An additional interface will be provided for the tensorcores AOT backend. 
```

Comment:
```suggestion
IMPORTANT: Since the current Tensor Cores  implementation is AOT, it is possible to know how many elements are owned by each WI at compile time. In this case, `wi_data` can be of type `marray`. An additional interface will be provided for the Tensor Cores AOT backend. 
```

Actually there are some aspects of the Tensor Cores backend that are not AOT, since there is JIT compilation from the ptx level to SASS level when the code is run. I think this would be more accurate and simpler personally:

```suggestion
IMPORTANT: It is possible to know how many elements are owned by each WI at compile time. In this case, `wi_data` can be of type `marray`. An additional interface will be provided for the Tensor Cores backend. 
```

Reply:
AOT is used here to reflect the SYCL compilation flow: If it does not use SPIRV and target is specified at compile time so this is basically AOT compilation for SYCL. But I will change the sentence to the one that does not talk about the AOT detail.